pkgs/tools/networking/pritunl-client/default.nix

```diff
@@ -1,27 +1,144 @@
-{ lib, buildGoModule, fetchFromGitHub }:
-
-buildGoModule rec {
-  pname = "pritunl-client";
+{ lib
+, stdenv
+, fetchFromGitHub
+, runtimeShell
+, runCommand
+, makeWrapper
+, installShellFiles
+, buildGoModule
+, coreutils
+, which
+, gnugrep
+, gnused
+, openresolv
+, systemd
+, iproute2
+, openvpn
+, electron
+, gcc-unwrapped
+}: let
   version = "1.3.3484.2";
-
   src = fetchFromGitHub {
     owner = "pritunl";
     repo = "pritunl-client-electron";
     rev = version;
     sha256 = "sha256-thIbw/Iek2vNNmkOBHrzEhTbaOm05CsbjqORQKC2jIs=";
   };
 
-  modRoot = "cli";
-  vendorHash = "sha256-miwGLWpoaavg/xcw/0pNBYCdovBnvjP5kdaaGPcRuWk=";
+  cli = buildGoModule {
+    pname = "pritunl-cli";
+    inherit version src;
+
+    modRoot = "cli";
+    vendorHash = "sha256-miwGLWpoaavg/xcw/0pNBYCdovBnvjP5kdaaGPcRuWk=";
+
+    postInstall = ''
+      mv $out/bin/cli $out/bin/pritunl-client
+    '';
+  };
+
+  service = buildGoModule rec {
+    pname = "pritunl-client-service";
+    inherit version src;
+
+    modRoot = "service";
+    vendorHash = "sha256-VuJ1giVSDlVFRJOy0UPVYp8j6TCWzXbiTgg4xyMq3dQ=";
+
+    nativeBuildInputs = [makeWrapper];
+
+    buildInputs = [
+      coreutils
+      which
+      gnused
+      gnugrep
+    ] ++ lib.optionals stdenv.isLinux [
+      openresolv
+      systemd
+      iproute2
+    ];
+
+    postPatch = ''
+      sed -i -E 's|#!\s*(/usr)?/bin/(env )?bash\b|#! ${runtimeShell}|g' service/profile/scripts.go
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      sed -i -E 's|(/usr)?/s?bin/busctl\b|busctl|g' service/profile/scripts.go
+      sed -i -E 's|(/usr)?/s?bin/resolvectl\b|resolvectl|g' service/profile/scripts.go
+      sed -i -E 's|(/usr)?/s?bin/ip\b|ip|g' service/profile/scripts.go
+    '';
+
+    postInstall = ''
+      mv $out/bin/service $out/bin/pritunl-client-service
+    '' + lib.optionalString stdenv.isLinux ''
+      mkdir -p $out/lib/systemd/system/
+      cp $src/resources_linux/pritunl-client.service $out/lib/systemd/system/
+    '';
+
+    postFixup = let
+      openvpn-wrapped = runCommand "openvpn-wrapped" {nativeBuildInputs = [makeWrapper];} ''
+        mkdir -p $out/bin
+        makeWrapper ${openvpn}/bin/openvpn $out/bin/openvpn \
+          --prefix PATH : ${lib.makeBinPath buildInputs} \
+          --add-flags "--setenv PATH \$PATH"
+      '';
+    in lib.optionalString stdenv.isLinux ''
+      wrapProgram $out/bin/pritunl-client-service \
+        --prefix PATH : "${lib.makeBinPath ([openvpn-wrapped] ++ buildInputs)}"
```

Comment:
```suggestion
        --prefix PATH : "${lib.makeBinPath ([ openvpn-wrapped ])}"
```
this would be wrong

Reply:
As I mentioned, just wanted to provide the same utils to both for consistency, because it's the service which hardcode the hook scripts and refers to the shell utils, but yes, it's only strictly required by openvpn (which run said scripts).